src/main/java/org/folio/rest/impl/VendorHelper.java

```diff
@@ -150,6 +153,24 @@ private Error createErrorWithId(ErrorCodes errorCodes, String id) {
     return error;
   }
 
+  /**
+   * Creates {@link Error} with id of corresponding vendor/access provider and poLines ids
+   *
+   * @param id vendor's/access provider's identifier
+   * @param errorCodes error code
+   * @param poLines list of composite PoLines
+   * @return {@link Error} with id of failed vendor/access provider
+   */
+  private Error createErrorWithId(ErrorCodes errorCodes, String id, List<CompositePoLine> poLines) {
+    Error error = createErrorWithId(errorCodes, id);
+    for (CompositePoLine poLine : poLines) {
+      if (poLine.getPoLineNumber() != null) {
```

Comment:
This logic does not take into account if the PO Line's access provider id equals to vendor `id` but just adds all PO Line numbers as parameters.
I verified this by `PurchaseOrdersApiTest#testPutOrdersByIdToChangeStatusToOpenInactiveVendor` unit test adding in the beginning of the test
```java
    for (int i = 0; i < reqData.getCompositePoLines().size(); i++) {
      reqData.getCompositePoLines().get(i).setPoLineNumber("number-" + i);
    }
```
After updates above the test fails when there is one active and one inactive access provider specified